app/javascript/packages/document-capture/components/acuant-capture.tsx

```diff
@@ -20,79 +23,107 @@ import DeviceContext from '../context/device';
 import UploadContext from '../context/upload';
 import useCounter from '../hooks/use-counter';
 import useCookie from '../hooks/use-cookie';
+import type {
+  AcuantSuccessResponse,
+  AcuantDocumentType,
+  AcuantCaptureFailureError,
+} from './acuant-camera';
 
-/** @typedef {import('react').ReactNode} ReactNode */
-/** @typedef {import('./acuant-camera').AcuantSuccessResponse} AcuantSuccessResponse */
-/** @typedef {import('./acuant-camera').AcuantDocumentType} AcuantDocumentType */
-/** @typedef {import('@18f/identity-components').FullScreenRefHandle} FullScreenRefHandle */
-/** @typedef {import('../context/acuant').AcuantGlobal} AcuantGlobal */
+type AcuantDocumentTypeLabel = 'id' | 'passport' | 'none';
+type AcuantImageAssessment = 'success' | 'glare' | 'blurry';
+type ImageSource = 'acuant' | 'upload';
 
-/**
- * @typedef {"id"|"passport"|"none"} AcuantDocumentTypeLabel
- */
-
-/**
- * @typedef {"success"|"glare"|"blurry"} AcuantImageAssessment
- */
-
-/**
- * @typedef {"acuant"|"upload"} ImageSource
- */
-
-/**
- * @typedef ImageAnalyticsPayload
- *
- * @prop {number?} width Image width, or null if unknown.
- * @prop {number?} height Image height, or null if unknown.
- * @prop {string?} mimeType Mime type, or null if unknown.
- * @prop {ImageSource} source Method by which image was added.
- * @prop {number=} attempt Total number of attempts at this point.
- * @prop {number} size Size of image, in bytes.
- */
+interface ImageAnalyticsPayload {
+  /**
+   * Image width, or null if unknown
+   */
+  width: number | null;
+  /**
+   * Image height, or null if unknown
+   */
+  height: number | null;
+  /**
+   * Mime type, or null if unknown
+   */
+  mimeType: string | null;
+  /**
+   * Method by which the image was added
+   */
+  source: ImageSource;
+  /**
+   * Total number of attempts at this point
+   */
+  attempt?: number;
+  /**
+   * Size of the image in bytes
+   */
+  size: number;
+}
 
-/**
- * @typedef _AcuantImageAnalyticsPayload
- *
- * @prop {AcuantDocumentTypeLabel} documentType
- * @prop {number} dpi
- * @prop {number} moire
- * @prop {number} glare
- * @prop {number} glareScoreThreshold
- * @prop {boolean} isAssessedAsGlare
- * @prop {number} sharpness
- * @prop {number} sharpnessScoreThreshold
- * @prop {boolean} isAssessedAsBlurry
- * @prop {AcuantImageAssessment} assessment
- */
+interface _AcuantImageAnalyticsPayload {
+  documentType: AcuantDocumentTypeLabel;
+  dpi: number;
+  moire: number;
+  glare: number;
+  glareScoreThreshold: number;
+  isAssessedAsGlare: boolean;
+  sharpness: number;
+  sharpnessScoreThreshold: number;
+  isAssessedAsBlurry: boolean;
+  assessment: AcuantImageAssessment;
+}
 
-/**
- * @typedef {ImageAnalyticsPayload & _AcuantImageAnalyticsPayload} AcuantImageAnalyticsPayload
- */
+type AcuantImageAnalyticsPayload = ImageAnalyticsPayload & _AcuantImageAnalyticsPayload;
```

Comment:
This is a good one-to-one port, but I _believe_ it was implemented the way it was previously due to limitations of a good "extends" mechanism which would be available to us in TypeScript proper, i.e.

```ts
interface AcuantImageAnalyticsPayload extends ImageAnalyticsPayload {
  documentType: AcuantDocumentTypeLabel;
  // ...
}
```

What do you think about making that revision?